code/harness/DiskContents.cpp

```diff
@@ -0,0 +1,298 @@
+#include "DiskContents.h"
+using std::endl;
+
+namespace fs_testing {
+
+fileAttributes::fileAttributes() {
+  dir_attr = NULL;
+  stat_attr = NULL;
+  md5sum = "";
+}
+
+fileAttributes::~fileAttributes() {
+}
+
+void fileAttributes::set_dir_attr(struct dirent* a) {
+  dir_attr = (struct dirent *) malloc(sizeof(struct dirent));
```

Comment:
If you are copying all the fields of the struct I think you can just use the `=` operator